packages/graphql/src/translate/translate-top-level-match.ts

```diff
@@ -106,3 +113,41 @@ export function createMatchClause({
 
     return matchQuery;
 }
+
+function createFulltextMatchClause(
+    matchNode: Cypher.NamedNode,
+    whereInput: GraphQLWhereArg | undefined,
+    node: Node,
+    context: Context
+): Cypher.db.FullTextQueryNodes {
+    const phraseParam = new Cypher.Param(context.resolveTree.args.phrase);
+    const scoreVar = context.fulltextIndex.scoreVariable;
+
+    const matchQuery = new Cypher.db.FullTextQueryNodes(
+        matchNode,
+        context.fulltextIndex.name,
+        phraseParam,
+        undefined,
+        scoreVar
+    );
+
+    const labelsChecks = node.getLabels(context).map((label) => {
+        return Cypher.in(new Cypher.Literal(label), Cypher.labels(matchNode));
+    });
```

Comment:
Alternatively:

```suggestion
    const expectedLabels = node.getLabels(context);
    const labelsChecks=matchNode.hasLabels(...expectedLabels);
```

This is a relatively new feature of the cypherBuilder, but should work fine and much easier :)